etc/cmake/options.cmake

```diff
@@ -33,13 +33,15 @@ option(OT_APP_RCP "enable RCP app" ON)
 option(OT_FTD "enable FTD" ON)
 option(OT_MTD "enable MTD" ON)
 option(OT_RCP "enable RCP" ON)
+option(OT_TCP "enable TCP" ON)
```

Comment:
We already have `OT_TCP` defined as `ot_option` in this file?

Reply:
Yes, I think the issue is `OPENTHREAD_CONFIG_TCP_ENABLE` default is 1 in code, but default OFF in CMake:
https://github.com/openthread/openthread/blob/6bfcc0d7d495fcc7bcf1cdd4070bee14c391ef95/src/core/config/ip6.h#L172

We should set `OT_TCP` default `ON` in `ot_option` below instead.

Reply:
I am not sure about setting `OT_TCP` to be `ON` by default in Cmake.

In general, all CMake `ot_otion`s are by default set to empty (not-specified) so to allow us to fall back to `OPENTHREAD_CONFIG_{}` values (which may be provided in a project/product-specific config file).

Reply:
Yeah, my point is that the reason why the build were failing before is that OpenThread core defaults to assume TCP it should support TCP. However, linking the tcplp library was disabled by default.

If we want to separately control OpenThread TCP support and linking the tcplp library, then we should probably have separate build configs.